drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -59,11 +60,11 @@ impl Handler {
         &mut self,
         tx_prev: bool,
         iostat: IoStatus,
-        rx_buf: &[u8],
-        tx_buf: &mut [u8],
+        rx_buf: &RxMsg,
+        rx_bytes: usize,
```

Comment:
Dumb question: should `rx_bytes` live in the `RxMsg` (indicating what subset of the buffer is valid), or is it independent?

Reply:
It's not a dumb question at all. I very much considered doing this, but it felt wrong, since RxMsg owns the buffer and would require a reset on the next transaction. Right now all the "size" information lives outside this type. 

I will think more about it as I go through the rest of the comments and refactor.